extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/BloomFilterAggregatorFactory.java

```diff
@@ -89,27 +87,69 @@ public BloomFilterAggregatorFactory(
   @Override
   public Aggregator factorize(ColumnSelectorFactory columnFactory)
   {
-    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
-        DimensionHandlerUtils.createColumnSelectorPlus(
-            STRATEGY_FACTORY,
-            field,
-            columnFactory
-        );
+    BloomKFilter filter = new BloomKFilter(maxNumEntries);
+    ColumnCapabilities capabilities = columnFactory.getColumnCapabilities(field.getDimension());
+    BaseNullableColumnValueSelector selector = columnFactory.makeColumnValueSelector(field.getDimension());
 
-    return new BloomFilterAggregator(selectorPlus, maxNumEntries);
+    if (selector instanceof NilColumnValueSelector) {
+      return new NilBloomFilterAggregator((NilColumnValueSelector) selector, filter);
```

Comment:
Is it important that `BloomKFilter` has specifically `maxNumEntries`? If not, `new BloomKFilter(0)` (or 1) could be cached in a constant.

Reply:
In any case, `(NilColumnValueSelector) selector` shouldn't be passed down, `NilColumnValueSelector` could call `super(NilColumnValueSelector.instance())` in its constructor.

Reply:
`BloomKFilter` must be the same size to merge, so it's not possible to make a constant. :+1: on the signature change though, will do that.

Reply:
Ok, please add a comment noting this.